core/lib/zksync_core/src/api_server/web3/pubsub.rs

```diff
@@ -388,3 +445,17 @@ impl EthSubscribe {
         notifier_tasks
     }
 }
+
+#[async_trait::async_trait]
+impl EthPubSubServer for EthSubscribe {
+    async fn subscribe(
+        &self,
+        pending: PendingSubscriptionSink,
+        sub_type: String,
+        filter: Option<PubSubFilter>,
+    ) -> SubscriptionResult {
+        self.sub(pending, sub_type, filter).await;
+
+        Ok(())
```

Comment:
Dumb question: Does `jsonrpsee` make any assumptions as to when this method will be resolved (e.g., is `PendingSubscriptionSink` required to be accepted / rejected?). In any case, IMO, it'd make sense to resolve the sync *before* spawning a Tokio task for it.

Reply:
resolving moved before spawning